### ISSUE TITLE
Node issue post upgrade from v1.16.0 to saffron v1.17.0

### ISSUE THREAD

**Issue by Palak3196** [provenance-io/provenance#1764]
Hi Team,

We have been running an archival provenance node with docker and have been updating it accordingly over the past year.

Our node stopped syncing at block 13736000, using image: provenanceio/provenance:v1.16.0
At this block, we went for upgrade to Saffrom v.1.17.0:

- Stopped the existing node container.
- Started it back again with command: `docker run -d --rm -it --name node-prov -p 1317:1317 -p 26657:26657 -p 9090:9090 -v /DataDisk/provenance-arch-node:/home/provenance provenanceio/provenance:v1.17.0 provenanced start`

But the node logs are stuck at this for the past 48 hours:
![image](https://github.com/provenance-io/provenance/assets/109664826/643b4196-ee40-48b9-b5d2-82f4cebc7a13)

Please let us know if any changes are needed to the docker command we're using.

Thank you for supporting us for more than a year now!

Palak

**Comment by nullpointer0x00** [comment]
Hi Palak,

Do you happen to know what database your node is using.  Is it the `cleveldb` or `goleveldb`?

**Comment by Palak3196** [comment]
@nullpointer0x00 Nope, Is there a way to find this out?

**Comment by Palak3196** [comment]
Hi Team, Still waiting for help here. Our node is still down

**Comment by Taztingo** [comment]
@Palak3196 We have a dbmigrate tool in the Provenance repo. I would recommend backing up your $PIO_HOME/data and trying that out.

You can build and install `dbmigrate` with the following commands
`make build-dbmigrate`
`make install-dbmigrate`

You can then run the `dbmigrate` command with `goleveldb` set as your target.
Make sure $PIO_HOME is the home directory you wish to migrate otherwise include the `--home` flag.
`dbmigrate goleveldb`

This should hopefully update your databases and allow you to continue running.

**Comment by Palak3196** [comment]
sounds perfect, let me try. Thank you for the quick support

**Comment by Palak3196** [comment]
Hi @Taztingo, do I need to have provenanced installed for the dbmigrate command to run? Because it doesn't run if I simply docker exec -it inside the container.

And yes the config and data directories exist at /DataDisk/provenance-arch-node

**Comment by Taztingo** [comment]
You may want to run `whereis provenanced` to find the exact location of the binary and verify that is part of your PATH.

**Comment by Palak3196** [comment]
Hi, unfortunately, whereis provenanced also gives no response
![image](https://github.com/provenance-io/provenance/assets/109664826/c9c57213-de45-4aa6-80c6-2a5d43ff1e7f)

I have tried the installation on Ubuntu 18.04 and 20.04.

**Comment by iramiller** [comment]
This means that you do not have your `GOBIN` in your path ... this is the folder where go will install binaries built with `make build`
```
%%> echo $GOBIN
/Users/[...]/Projects/go/bin
```
and
```
 %%> echo $PATH
/opt/homebrew/opt/libpq/bin /opt/homebrew/bin /opt/homebrew/sbin /opt/homebrew/opt/fzf/bin /Users/[...]/Projects/go/bin [...]
```
This is why I am able to resolve my installed `provenanced` binary like so
```
%%> which provenanced
/Users/[...]/Projects/go/bin/provenanced
```

**Comment by Palak3196** [comment]
Thanks @iramiller  I added GOBIN in the path as suggested:
![image](https://github.com/provenance-io/provenance/assets/109664826/b15bbf17-1045-415b-a060-2fedb05d7ec8)

And then ran make build in the provenance binaries' directory (taken from v 1.17.0), but it ran into errors (attached in the note)
[Prov error.txt](https://github.com/provenance-io/provenance/files/13943136/Prov.error.txt)


Also, I saw the release notes for Release v1.17.1 and tried the below steps too to goleveldb from cleveldb:
Stop your node.
Back up your data and config directories.
Update your config.toml to have db_backend = "goleveldb".
Update your app.toml to have either app-db-backend = "" or app-db-backend = "goleveldb".
Restart your node.

For this, my docker container runs into the below error:
![image](https://github.com/provenance-io/provenance/assets/109664826/841080ae-91b5-4c6e-96ac-caf51f7c5907)

Sorry that's a lot of information! Let me know if there's anything wrong here
